### ISSUE TITLE
torch.fill_() exists and modifies the input tensor: Expected or bug?

### ISSUE THREAD

**Issue by xuhdev** [pytorch/pytorch#22707]
Seems like in-place functions are method only, but torch.fill_() exists, is undocumented, but is declared in both native_functions.yml and derivatives.yaml.

    import torch
    a = torch.zeros(3)
    torch.fill_(a, 1)
    print(a)

This prints:

    tensor([1., 1., 1.])

Is this expected?

- PyTorch Version (e.g., 1.0): master

**Comment by zou3519** [comment]
We should probably kill it, only methods should be in-place.

**Comment by SsnL** [comment]
probably should filter at https://github.com/pytorch/pytorch/blob/1ecc945ab23e54eabcd5db1d4bbc779abd908e49/tools/autograd/gen_python_functions.py#L233

**Comment by xuhdev** [comment]
Does this require some discussion? I'm a bit worried that many users at large may have used it, like pytorch/vision#1107 . If we want to fix this, we should do it quickly (as started in #22720 ) to prevent more users from misusing them.

**Comment by SsnL** [comment]
This is a bug. In-place functions don't exist before (only methods exist), nor are they documented/advertised anywhere. We should just kill them.

**Comment by zou3519** [comment]
I wonder if it's possible to prevent a function from appearing in the torch binding unless it is documented... this way we can avoid problems like this. I'm sure there are a lot of other functions that are undocumented and (probably) erroneously exposed to torch.*

**Comment by gchanan** [comment]
> I wonder if it's possible to prevent a function from appearing in the torch binding unless it is documented... this way we can avoid problems like this. I'm sure there are a lot of other functions that are undocumented and (probably) erroneously exposed to torch.*

For this specific case we should probably just throw a quick check in the codegen.

**Comment by gchanan** [comment]
Added needs discussion label based on the request of @xuhdev over in https://github.com/pytorch/pytorch/pull/22720.

My personal opinion is we should accept a patch that does the proper deprecation.

**Comment by ezyang** [comment]
In triage meeting, we accepted Greg's opinion. Let's do a proper deprecation cycle on these functions. Note that a lot of functions need this treatment too: functions that are bound in torch that should not be bound there. (For example, `conv2d`). @xuhdev, does this make sense?

**Comment by xuhdev** [comment]
Makes sense. There is no absolutely right way to deal with these kinds of issues; I'll defer the judgement to the people who know more about this part (and the future plan on it).
